packages/react-shopify-app-bridge/spec/useGadget.spec.ts

```diff
@@ -0,0 +1,17 @@
+import { AppBridgeState, ClientApplication } from "@shopify/app-bridge";
+import { assert, IsExact } from "conditional-type-checks";
+import { AppType, useGadget } from "../src";
+
+// these functions are typechecked but never run to avoid actually making API calls
+const TestUseGadgetReturnsAppropriateTypes = () => {
+  const { loading, appType, isEmbedded, appBridge, canAuth, isAuthenticated } = useGadget();
+
+  assert<IsExact<typeof loading, boolean>>(true);
+  assert<IsExact<typeof isEmbedded, boolean>>(true);
+  assert<IsExact<typeof canAuth, boolean>>(true);
+  assert<IsExact<typeof isAuthenticated, boolean>>(true);
+  assert<IsExact<typeof appType, AppType | undefined>>(true);
+  assert<IsExact<typeof appBridge, ClientApplication<AppBridgeState> | undefined>>(true);
+};
```

Comment:
Not sure this test really provides any value with maybe the exception of the `| undefined` assertions as we already have some lint checks.

Reply:
Yeah, TypeScript should take care of most of this, but it could be helpful to catch possible problems casting through any?